accepted/00000-coco-attestation.md

```diff
@@ -0,0 +1,305 @@
+- Feature Name: Confidential Compute Attestation Report
+- Start Date: 2024-01-16
+
+# Summary
+[summary]: #summary
+
+SUSE Manager / Uyuni should help with Confidential Compute Attestation and being able to generate a report page for systems which should run in this mode.
+
+# Motivation
+[motivation]: #motivation
+
+More and more systems run as virtual machines or (public) cloud instances.
+Customers want to be sure that data of these VMs cannot be read by administrators or the cloud provider.
+Confidential Compute is the way to solve this problem.
+But that a VM run in confidential mode needs to be attested and checked on regular base.
+
+SUSE Manager should provide an attestation report for a system.
+Also a history of the past checks should be kept and shown.
+
+The attestation is a process of multiple checks which can also depend on the hardware and used environment.
+SUSE Manager should provide a framework for executing such checks, store the results in the database and show the latest and historic results.
+The number of checks should be easy extendable in future versions.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Confidential Compute Attestation cannot be performed on all systems.
+It depends on the used hardware and environment where the instance is running on.
+In most cases it is not possible to detect if an attestation can be performed or not.
+
+This requires a per system configuration which should contain at least two configuration options:
+
+- Enable/Disable Confidential Compute Attestation (Checkbox)
+- Environment selection
+  - KVM Guest Attestation on AMD EPYC Milan CPUs
+  - KVM Guest Attestation on AMD EPYC Genoa CPUs
+  - Azure Attestation
+  - etc.
+
+The information about the environment result in pillar data which help to select the matching states which should be executed on the minion for this task.
+
+The attestation is a 2 step process.
+First collect the data and second to verify them and write the result.
+
+### Step One
+The first step execute a state on the target system to collect the report data.
```

Comment:
Considering this would be an async call, it does make sense to have this additional status.

Reply:
Remember that we also have an action. This hold that state. I think "pending" is enough here.
But we may need a pointer to the action

Reply:
We need something to know that the report/Result is ready to be processed. It can be a new state in the report table or a pointer to the action table. 
The result attestation container(s) needs to know the action is already processed and data should be available.

Reply:
The idea is, that we generate the "result" entries with status `pending` and the containers would just look on pending entries in the result table, not in the report table.

Reply:
The plan is to create them when salt return and we evaluate the result. 

`At the end of step one all the expected result checks are created` is part of this section now